tests/unit/helpers/test_recon_config_utils.py

```diff
@@ -143,3 +153,112 @@ def test_store_connection_secrets_overwrite(mock_workspace_client):
+
+
+def test_generate_recon_config_no_secrets_configured(mock_workspace_client):
+    prompts = MockPrompts(
+        {
+            r"Select the source": SOURCE_DICT["snowflake"],
+            r".*Did you setup the secrets for the": "no",
+        }
+    )
+
+    recon_conf = ReconConfigPrompts(mock_workspace_client, prompts)
+    recon_conf.prompt_source()
+
+    error_msg = (
+        "Error: Secrets are needed for `snowflake` reconciliation.\n"
+        "Use `remorph configure-secrets` to setup Scope and Secrets."
+    )
+
+    with pytest.raises(ValueError, match=error_msg):
+        recon_conf.prompt_and_save_config_details()
+
+
+def test_generate_recon_config_create_scope_no(mock_workspace_client):
+    prompts = MockPrompts(
+        {
+            r"Select the source": SOURCE_DICT["snowflake"],
+            r".*Did you setup the secrets for the": "yes",
+            r"Enter Secret Scope name": "dummy",
+            r"Do you want to create a new one?": "no",
+        }
+    )
+
+    recon_conf = ReconConfigPrompts(mock_workspace_client, prompts)
+    recon_conf.prompt_source()
+
+    error_msg = "Scope is needed to store Secrets in Databricks Workspace"
+
+    with pytest.raises(SystemExit, match=error_msg):
+        recon_conf.prompt_and_save_config_details()
+
+
+def test_recon_config_prompt_and_save_config_details(mock_workspace_client):
+    filter_dict = {"exclude": "0", "include": "1"}
+
+    prompts = MockPrompts(
+        {
+            r"Select the source": SOURCE_DICT["snowflake"],
+            r".*Did you setup the secrets for the": "yes",
+            r"Enter Secret Scope name": "dummy",
+            r"Enter `snowflake` catalog_name": "sf_catalog",
+            r"Enter `snowflake` database_name": "sf_schema",
+            r"Enter target catalog_name": "tgt_catalog",
+            r"Enter target schema_name": "tgt_schema",
+            r"Do you want to include/exclude a set of tables?": "yes",
+            r"Select the filter type": filter_dict["include"],
+            r"Enter the tables.*": "table1, table2",
+            r"Open.* config file in the browser?": "yes",
+        }
+    )
+
+    filename = "recon_conf_snowflake.json"
+    recon_confing = TableRecon(
+        source_schema="src_schema",
+        source_catalog="src_catalog",
+        target_catalog="tgt_catalog",
+        target_schema="tgt_schema",
+        tables=[Table(source_name="table1", target_name="table1"), Table(source_name="table2", target_name="table2")],
+    )
+
+    # Patch the scope_exists method to return True
+    mock_workspace_client.secrets.list_scopes.side_effect = [[SecretScope(name="dummy")]]
+    # Patch the builder method to return a SparkSession
+    with patch.object(DatabricksSession, "builder", MagicMock(return_value=SparkSession.builder)):
+        # mock WorkspaceClient and Installation in _save_config_details
+        with patch("databricks.labs.remorph.helpers.recon_config_utils.WorkspaceClient", mock_workspace_client):
```

Comment:
Obscure implicit test dependency with mock.patch(`databricks....`). Rewrite to inject dependencies through constructor.. Using patch to mock dependencies in unit tests can introduce implicit dependencies within a class, making it unclear to other developers. Constructor arguments, on the other hand, explicitly declare dependencies, enhancing code readability and maintainability. However, reliance on patch for testing may lead to issues during refactoring, as updates to underlying implementations would necessitate changes across multiple unrelated unit tests. Moreover, the use of hard-coded strings in patch can obscure which unit tests require modification, as they lack strongly typed references. This coupling of the class under test to concrete classes signifies a code smell, and such code is not easily portable to statically typed languages where monkey patching isn't feasible without significant effort. In essence, extensive patching of external clients suggests a need for refactoring, with experienced engineers recognizing the potential for dependency inversion in such scenarios.

To address this issue, refactor the code to inject dependencies through the constructor. This approach explicitly declares dependencies, enhancing code readability and maintainability. Moreover, it allows for dependency inversion, enabling the use of interfaces to decouple the class under test from concrete classes. This decoupling facilitates unit testing, as it allows for the substitution of mock objects for concrete implementations, ensuring that the class under test behaves as expected. By following this approach, you can create more robust and maintainable unit tests, improving the overall quality of your codebase.

https://github.com/databrickslabs/pylint-plugin?tab=readme-ov-file#r8918-explicit-dependency-required